[Accelerate] [vDSP] Merge, Swap, Gather, and Compress Functions

FlexMonkey opened the pull request:
There were a handful of overlooked functions in my first batch of work. This PR adds shims to: vDSP_vgenp, vDSP_vtabi, vDSP_vtmerg, vDSP_vswap, vDSP_vgathr, vDSP_vcmprs.

These aren't frequently used operations, so there's no mad rush. I have run the tests without the `REQUIRES` tag, and everything works as expected.

This PR also includes a new initialiser for vImage buffers from `CGSize`.

cc: @stephentyrone @airspeedswift

Commit:
[Accelerate] [vDSP] Merge, Swap, Gather, and Compress Functions

Add shims to: vDSP_vgenp, vDSP_vtabi, vDSP_vtmerg, vDSP_vswap, vDSP_vgathr, vDSP_vcmprs

Commit:
Add Init From CGSize to vImage Buffer

As per conversation - I've added a new init to `vImage_Buffer` from a `CGSize`. This uses `exactly` to create the integer sizes and throws on failure.

Commit:
Undelete "Requires" Tags in Test

stephentyrone commented:
@swift-ci please test

stephentyrone commented:
@FlexMonkey looks like this needs updating to land on master. Can you take a look, and let me know if you need help with it?

FlexMonkey commented:
Will do - looking right now.

Thanks!

simon

Simon Gladman     +44 7973 669691

Blog: http://flexmonkey.blogspot.co.uk
GitHub: https://github.com/FlexMonkey
Twitter: @FlexMonkey <https://twitter.com/FlexMonkey>



On Tue, 6 Aug 2019 at 17:06, Stephen Canon <notifications@github.com> wrote:

> @FlexMonkey <https://github.com/FlexMonkey> looks like this needs
> updating to land on master. Can you take a look, and let me know if you
> need help with it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apple/swift/pull/25180?email_source=notifications&email_token=ABPO34QPPJXU4GWQFKDN2VTQDGOR3A5CNFSM4HR6IOHKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3VUNHI#issuecomment-518735517>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPO34SFGUKJN3ISJLLXIPTQDGOR3ANCNFSM4HR6IOHA>
> .
>

FlexMonkey committed:
Merge branch 'master' into accelerate-vDSP-MergeSwapGatherCompress

FlexMonkey committed:
Update availaable versions.

FlexMonkey committed:
Fix @available

FlexMonkey committed:
Deprecate SplitComplex Inits

FlexMonkey committed:
Revert "Deprecate SplitComplex Inits"

This reverts commit 43f25c9a49ba0f9b5aadc4d22c7af2f6f739665e.

lorentey commented on stdlib/public/Darwin/Accelerate/vDSP_FillClearGenerate.swift:
```diff
@@ -731,5 +731,275 @@ extension vDSP {
+    /// - Parameter indices: The monotonically increasing indices to `result` for the corresponding element in `values`.
+    /// - Parameter result: Output values.
+    @inlinable
+    @available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
```
These need to be changed to `@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, *)` everywhere, because we don't know which OS versions will ship these yet.

FlexMonkey replied:
Hi, this is fixed now.

lorentey commented on stdlib/public/Darwin/Accelerate/vImage_Buffer.swift:
```diff
@@ -72,6 +72,34 @@ extension vImage_Buffer {
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init(size: CGSize,
```
This needs its own `@available` attribute, or it needs to be added to a new extension that has the correct availability. (It inherits `macOS 10.15`, `iOS 13`, etc. from its current context, which is wrong.)

FlexMonkey replied:
Hi, this is fixed now.

lorentey reviewed (changes requested):
I found some availability issues, but otherwise this looks good to me!

FlexMonkey committed:
Update Availability on New API

FlexMonkey committed:
Merge branch 'master' into accelerate-vDSP-MergeSwapGatherCompress

FlexMonkey committed:
Reintroduce deleted `REQUIRES`

`REQUIRES: rdar50301438` was temporarily deleted during development. This commit reintroduces them.

FlexMonkey committed:
Update overlooked availability

futurejones commented:
Can one of the admins verify this patch?

CodaFi commented:
@swift-ci test

stephentyrone commented:
Closing this for now.
